src/storable.rs

```diff
@@ -23,22 +23,20 @@ pub trait Storable {
     /// are within the element's bounds.
     fn to_bytes_checked(&self) -> Cow<[u8]> {
         let bytes = self.to_bytes();
-        if let Bound::Bounded {
-            max_size,
-            is_fixed_size,
-        } = Self::BOUND
-        {
-            if is_fixed_size {
-                assert_eq!(
-                    bytes.len(),
-                    max_size as usize,
+        let bound = Self::BOUND;
```

Comment:
nit: this is completely subjective, but I actually find the previous code more readable, as it was a simple match statement, so I'd prefer we keep this method as-is.